packages/react/src/create-strict-api/types.ts

```diff
@@ -0,0 +1,40 @@
+import type {
+  StrictCSSProperties,
+  CSSPseudoClasses,
+  CSSPseudoElements,
+  CSSPseudos,
+} from '../types';
+
+export type CompiledSchemaShape = StrictCSSProperties & {
+  [Q in CSSPseudoClasses]?: StrictCSSProperties;
+};
+
+export type PseudosDeclarations = { [Q in CSSPseudos]?: StrictCSSProperties };
+
+export type AllowedStyles = StrictCSSProperties & PseudosDeclarations;
+
+export type ApplySchemaValue<
+  TSchema,
+  TKey extends keyof StrictCSSProperties,
+  TPseudoKey extends CSSPseudoClasses | ''
+> = TKey extends keyof TSchema
+  ? TPseudoKey extends keyof TSchema
+    ? TKey extends keyof TSchema[TPseudoKey]
+      ? TSchema[TPseudoKey][TKey]
+      : TSchema[TKey]
+    : TSchema[TKey]
+  : StrictCSSProperties[TKey];
+
+export type ApplySchema<TObject, TSchema, TPseudoKey extends CSSPseudoClasses | '' = ''> = {
+  [TKey in keyof TObject]?: TKey extends keyof StrictCSSProperties
+    ? ApplySchemaValue<TSchema, TKey, TPseudoKey>
+    : TKey extends CSSPseudoClasses
+    ? ApplySchema<TObject[TKey], TSchema, TKey>
```

Comment:
This passes on the pseudo property key so we can use it to get the actual value. We'll most likely need to change this type when implementing nested pseudos like:

```
css({
  ':hover': {
    '::after': {}
  }
});
```

Or we could potentially keep them flat. Open to ideas — but for another PR not this one.

Reply:
I gather it's just another ternary nesting, eg. `Q in CSSPseudoClasses ? …` vs. `Q in CSSPseudoElements ? …`.  Doing my head in a bit, but same as doing all of this `TSchema[TPseudoKey][TKey]`…

It's fortunately only a maximum of 5 layers deep 😨 inside `CssMap`, but I think they're all explicitly different layers?  Something like?
```js
MapKey // primary
  CSSKey // display
  PseudoElementKey // &::after
    PseudoElementKey // &:hover
      CSSKey // display
  PseudoElementKey // &:hover
    CSSKey // display
    PseudoElementKey // &::after
      CSSKey // display
  AtRuleKey // @media
    PseudoElementKey // &:hover
      CSSKey // display
      PseudoElementKey // &::after
        CSSKey // display
    PseudoElementKey // &::after
      PseudoElementKey // &:hover
        CSSKey // display
    PseudoElementKey // &:hover
      CSSKey // display
```

Reply:
Yeah for sure. Something we can build on!